bundles/org.openhab.binding.radiothermostat/src/main/resources/OH-INF/addon/addon.xml

```diff
@@ -8,4 +8,41 @@
 	<description>Controls the RadioThermostat model CT30, CT50 or CT80 via the built-in WIFI module</description>
 	<connection>local</connection>
 
+	<discovery-methods>
+		<discovery-method>
+			<service-type>ip</service-type>
+			<discovery-parameters>
+				<discovery-parameter>
+					<name>type</name>
+					<value>ipMulticast</value>
+				</discovery-parameter>
+				<discovery-parameter>
+					<name>destIp</name>
+					<value>239.255.255.250</value>
+				</discovery-parameter>
+				<discovery-parameter>
+					<name>destPort</name>
+					<value>1900</value>
+				</discovery-parameter>
+				<discovery-parameter>
+					<name>request</name>
+					<value>0x54 0x59 0x50 0x45 0x3a 0x20 0x57 0x4d 0x2d 0x44 0x49 0x53 0x43 0x4f 0x56 0x45 0x52 0x0d 0x0a 0x56 0x45
+						0x52 0x53 0x49 0x4f 0x4e 0x3a 0x20 0x31 0x2e 0x30 0x0d 0x0a 0x0d 0x0a 0x73 0x65 0x72 0x76 0x69 0x63 0x65 0x73 0x3a
+						0x63 0x6f 0x6d 0x2e 0x6d 0x61 0x72 0x76 0x65 0x6c 0x6c 0x2e 0x77 0x6d 0x2e 0x73 0x79 0x73 0x74 0x65 0x6d 0x2a 0x0d
+						0x0a 0x0d 0x0a</value>
```

Comment:
I compared with the discovery implementation in the binding, and found the string behind these bytes. Do you think it would be possible to do something like this?
```suggestion
					<name>requestPlain</name>
					<value>TYPE: WM-DISCOVER\r\nVERSION: 1.0\r\n\r\nservices:com.marvell.wm.system*\r\n\r\n</value>
```

I didn't carefully check the [implementation](https://github.com/openhab/openhab-core/blob/593681a681a72e50a717efa31ac7dc3508c8cc79/bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java#L490-L511), but at least I think that is its intended use. If it doesn't work already, perhaps it should be tweaked to work. FYI @mherwege, @holgerfriedrich

Reply:
It definitely needs tweaking. The device requires all of the \r\n characters in the packet. However when requirestPlain parses the string, it does not un-escape these escaped characters and instead sends the characters individually: (ie: \ r \ n \ r \ n)
![image](https://github.com/openhab/openhab-addons/assets/26532542/529e96c9-5a26-4cab-8942-0b042519603c)

Reply:
Hmm, that seems like an XML issue. I wonder if `&#xD;&#xA;` would help or if `CDATA` could be used for the encoding.

Reply:
I think the core StringUtils.unEscapeXml() would need to be updated to include processing for `&#xD;` and `&#xA;`

https://github.com/openhab/openhab-core/blob/593681a681a72e50a717efa31ac7dc3508c8cc79/bundles/org.openhab.core/src/main/java/org/openhab/core/util/StringUtils.java#L322

Reply:
We could tweak ```buildRequestArrayPlain``` to unescape ```\x``` in ```requestPlain```. This should not break anything - at least in the official add-ons repo (```requestPlain``` is used by ipcamera and goovee, not backslash is used).
Maybe this behaviour is expeced by users anyway.

```String``` contains a method ```translateEscapes()``` which could be used to do this.

Reply:
Regex matching is on the TODO list for quite a while now, see openhab/openhab-core#3936. I am not working on the finders at the moment.
Just for completeness, could you comment to which add-on would need that functionality? Could you comment to openhab/openhab-core#3936? Thanks!